client-app/src/Components/DonorForm.tsx

```diff
@@ -2,6 +2,294 @@ import React, { useState, ChangeEvent, FormEvent } from 'react';
 import axios from 'axios';
 import '../css/DonorForm.css';
 
+interface FormData {
+    itemType: string;
+    currentStatus: string;
+    donorEmail: string;
+    program: string;
+    imageUpload: string[];
+    dateDonated: string;
+}
+
+interface FormErrors {
+    [key: string]: string;
+}
+
+const DonorForm: React.FC = () => {
+    const [formData, setFormData] = useState<FormData>({
+        itemType: '',
+        currentStatus: 'Received',
+        donorEmail: '',
+        program: '',
+        imageUpload: [],
+        dateDonated: '',
+    });
+
+    const itemTypeOptions = [
+        {value: 'bicycle', label: 'Bicycle'},
+        {value: 'computer', label: 'Computer'},
```

Comment:
More item types may be added for the next sprint as the user have more options to donate.